server/server_test.go

```diff
@@ -391,6 +422,65 @@ func makeOAuth2Tests(clientID string, clientSecret string, now func() time.Time)
 					return nil
 				},
 			},
+			{
+				name: "verify OIDC group prefixing on refresh token",
```

Comment:
Here we check if the `mock` prefix is also added on refresh. Also we check that the `groups` do not change after refresh to avoid a situation with extra prefixes.